src/pkg/storage/storage.go

```diff
@@ -166,13 +171,71 @@ func mustMatchConfig(
 		if !ok {
 			continue // m may declare values which aren't stored in the config file
 		}
-
 		vv := cast.ToString(g.Get(tomlK))
-		if v != vv {
+
+		areEqual := false
+
+		if IsValidPath(v) && IsValidPath(vv) {
+			areEqual, _ = ArePathsEquivalent(v, vv)
+		} else {
+			areEqual = v == vv
+		}
+
+		if !areEqual {
 			err := clues.New("value of " + k + " (" + v + ") does not match corso configuration value (" + vv + ")")
 			return clues.Stack(ErrVerifyingConfigStorage, err)
 		}
 	}
 
 	return nil
 }
+
+func ArePathsEquivalent(path1, path2 string) (bool, error) {
+	normalizedPath1 := strings.TrimSpace(filepath.Clean(path1))
+	normalizedPath2 := strings.TrimSpace(filepath.Clean(path2))
+
+	normalizedPath1 = strings.TrimSuffix(normalizedPath1, string(filepath.Separator))
+	normalizedPath2 = strings.TrimSuffix(normalizedPath2, string(filepath.Separator))
+
+	return normalizedPath1 == normalizedPath2, nil
+}
+
+func IsValidPath(path string) bool {
+	if _, err := os.Stat(path); err == nil {
+		return true
+	}
+
+	isDir := strings.TrimSpace(filepath.Ext(path)) == ""
+
+	if isDir {
+		err := os.Mkdir(path, 0644)
+		if err == nil {
+			os.Remove(path)
+			return true
+		}
+
+		if os.IsPermission(err) {
+			logger.Ctx(context.Background()).Info("directory could have been created but got permission error")
```

Comment:
avoid using context.Background() whenver possible.  Can we pass a ctx into this func?
```suggestion
			logger.Ctx(ctx).Info("permission restrictions prevented directory creation")
```